query/src/lib.rs

```diff
@@ -515,3 +516,79 @@ pub struct FindQuery {
     pub where_clauses: Vec<WhereClause>,
     // TODO: in_rules;
 }
+
+pub trait ContainsVariables {
+    fn acc_mentioned_variables(&self, acc: &mut BTreeSet<Variable>);
```

Comment:
Or `mentioned_variables`, and then make the other function `collect_mentioned_variables`.